platforms/core-configuration/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/TaskContainerExtensions.kt

```diff
@@ -164,15 +165,23 @@ private constructor(
      * @see [TaskProvider.configure]
      */
     operator fun String.invoke(configuration: Task.() -> Unit): TaskProvider<Task> =
-        this().apply { configure(configuration) }
+        named(this).apply { configure(configuration) }
 
     /**
      * Locates a task by name, without triggering its creation or configuration, failing if there is no such task.
      *
      * @see [TaskContainer.named]
      */
-    operator fun String.invoke(): TaskProvider<Task> =
-        container.named(this)
+    @Deprecated("Use named(String) instead.", ReplaceWith("named(\"...\")"))
+    operator fun String.invoke(): TaskProvider<Task> {
+        DeprecationLogger.deprecateBehaviour(String.format("Task '%s' found by String.invoke() notation.", this))
+            .withContext("The \"name\"() notation can cause confusion with methods provided by Kotlin or the JDK.")
```

Comment:
💭 Use backticks to indicate code?
```suggestion
            .withContext("The `\"name\"()` notation can cause confusion with methods provided by Kotlin or the JDK.")
```